### PR TITLE
fix: allow activeProps to infer component props

### LINKED ISSUE #2079: When using createLink(), activeProps type should allow component props
### Describe the bug

image

TS error: `
Object literal may only specify known properties, and 'selected' does not exist in type 'AnchorHTMLAttributes<HTMLAnchorElement> | (() => AnchorHTMLAttributes<HTMLAnchorElement>)'.
`

Beside, no `className` should be provided to component to make this function completely agnostic.

### Your Example Website or App

https://stackblitz.com/edit/github-ayifmj?file=src%2Fmain.tsx,src%2Froutes%2Findex.tsx

### Steps to Reproduce the Bug or Issue

Click on the stackblitz link and notice the TS error.

### Expected behavior

No TS error.

### Screenshots or Videos

_No response_

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chorobin**
Fixes #2079

### CODE REVIEW

File: packages/react-router/src/link.tsx [hunk 1]
```diff
@@ -846,50 +846,67 @@ export function useLinkProps<
...
+
 export type UseLinkPropsOptions<
   TRouter extends AnyRouter = RegisteredRouter,
   TFrom extends RoutePaths<TRouter['routeTree']> | string = string,
   TTo extends string | undefined = '.',
   TMaskFrom extends RoutePaths<TRouter['routeTree']> | string = TFrom,
   TMaskTo extends string = '.',
-> = ActiveLinkOptions<TRouter, TFrom, TTo, TMaskFrom, TMaskTo> &
-  React.AnchorHTMLAttributes<HTMLAnchorElement>
+> = ActiveLinkOptions<'a', TRouter, TFrom, TTo, TMaskFrom, TMaskTo> &
```
Comment (schiller-manuel) [inline packages/react-router/src/link.tsx:868]: about the `'a'`: please see this comment https://github.com/TanStack/router/pull/2092#discussion_r1707677038

Comment (chorobin) [reply packages/react-router/src/link.tsx:868]: This works with 'a' properly now ;)

Comment (schiller-manuel) [reply packages/react-router/src/link.tsx:868]: cool!
@mirague check this out

File: docs/framework/react/guide/custom-link.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        className="block py-2 px-3 text-blue-700"
+        preload="intent"
+        activeProps={{ className: `font-bold` }}
+      />
+    )
+  }),
+)
+```
+
+> React.forwardRef wil not be required in React 19 -->
```
Comment (schiller-manuel) [inline docs/framework/react/guide/custom-link.md:31]: wil => will

File: docs/framework/react/guide/custom-link.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    invoiceId: 0,
+  }}
+/>
+```
+
+# `createLink` with third party libraries
+
+You might want to use third party component libraries with TanStack Router. For example to use `Button` from MUI you can use `createLink` which infers the types from both `Button` and `Link` while keeping type parameters necessary for TanStack Router's type safety
+
+```tsx
```
Comment (schiller-manuel) [inline docs/framework/react/guide/custom-link.md:48]: i would add the imports for `createLink` and `Button` here

File: docs/framework/react/guide/custom-link.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+const ButtonLink = createLink(
+  React.forwardRef(
+    (
+      props: Parameters<typeof Button>[0],
+      ref: React.Ref<HTMLButtonElement>,
+    ) => {
+      return <Button {...props} ref={ref} component={Link} />
```
Comment (schiller-manuel) [inline docs/framework/react/guide/custom-link.md:55]: and how does this button then get acess to the `to`?

Comment (chorobin) [reply docs/framework/react/guide/custom-link.md:58]: Does it need to? Its there at runtime but its not been typed. I'm not sure the use case for this though

Comment (schiller-manuel) [reply docs/framework/react/guide/custom-link.md:58]: what if you want to navigate to `to` upon button click?

Comment (mirague) [reply docs/framework/react/guide/custom-link.md:58]: We have a similar usecase for getting access to the `active: boolean` property to render a different variant of the component based on it being active. 

_(`activeProps` doesn't seem to propagate as expected)_

**Pull request merged**